### PR TITLE
[GH-43213] Fix duplicate comments printed in quick info section

### LINKED ISSUE #43213: Duplicate JSDoc in union of derived interfaces
```ts
interface I {
    /** The colour */
    readonly colour: string
}
interface A extends I {
    readonly colour: "red" | "green";
}
interface B extends I {
    readonly colour: "yellow" | "green";
}
type F = A | B
const f: F = { colour: "green" }
f.colour/**/
```

**Expected behavior:**

Quickinfo at `/**/`
```
(property) I.colour: string

The colour
```

**Actual behavior:**

```
(property) I.colour: string

The colour
The colour
```

From my notes on discord explaining to the person who noticed this: 

The function in the TS services is called getDocumentationComment, and it does three things:
1. calls another function to get  the comment text to display
2. loops over each declaration after uniquifying
3. per-declaration, gets the comment text from the base, if there is any

For the example above, colour has two declarations, since it's a union: Apple.colour and Banana.colour. So: 

Step 1. No comment text from either declaration.
Step 2. There are two unique declarations to iterate over.
Step 3. Each declaration has the same base, which contributes the same comment.

#35170 has separate jsdocs for each, as you point out.
#32708 only has one extended declaration, just instantiated with two generic types

### PR DESCRIPTION
**Pull request by hahmadia**
Fixes #43213

Only push values to the comments array if it's unique (similar to how we do it for declartation comments in         let doc = JsDoc.getJsDocCommentsFromDeclarations)

### CODE REVIEW

File: src/services/services.ts [hunk 1]
```diff
@@ -584,7 +584,17 @@ namespace ts {
             forEachUnique(declarations, declaration => {
                 const inheritedDocs = findBaseOfDeclaration(checker, declaration, symbol => symbol.getDocumentationComment(checker));
                 // TODO: GH#16312 Return a ReadonlyArray, avoid copying inheritedDocs
-                if (inheritedDocs) doc = doc.length === 0 ? inheritedDocs.slice() : inheritedDocs.concat(lineBreakPart(), doc);
```
Comment (sandersn) [inline src/services/services.ts:587]: I don't think you need to write your own loop; `findBaseOfDeclaration` allows you to skip base declarations by returning undefined, so you can maintain `seen: Set<Symbol>` to allow you to skip a symbol that's been seen before.

Comment (sandersn) [reply src/services/services.ts:587]: Also, the outer forEachUnique might be redundant with this fix in place. Might be worth running tests once with it removed to see.

**Review by sandersn — CHANGES_REQUESTED** [review]
An idea for a simpler approach.

File: src/services/services.ts (hunk 1 above)
Comment (hahmadia) [reply src/services/services.ts:587]: GENUIS!

**Comment by hahmadia** [comment]
@sandersn 
I've addressed your comments.

**Comment by hahmadia** [comment]
I removed the forEachUnique since it wasn't really doing much in this case and replaced it with a normal for each loop.

**Review by sandersn — APPROVED** [review]

**Pull request merged**